packages/components/FocusZone/macos/RCTFocusZone.m

```diff
@@ -480,6 +490,10 @@ - (void)keyDown:(NSEvent *)event
 		[[self window] makeFirstResponder:viewToFocus];
 		[viewToFocus scrollRectToVisible:[viewToFocus bounds]];
 	}
+	else if (noOp)
+	{
+		// No view to focus, do nothing
+	}
```

Comment:
I don't like the idea of `noOp`big here. It prevents the keyboard event from being handled by `super` which means something else in the responder chain (superviews, the window, open menus) won't be able to handle the event. That seems like bad practice to me.

Reply:
Is there a callstack showing why we beachball? What causes the infinite loop when invoking `super`?

Reply:
^ that's my question. What's the actual infinite loop that is happening? Is `nextValidKeyView == self` ?

Reply:
@Saadnajmi  I feel like FocusZone as a feature overrides the native keyboarding behavior to custom fit our business requirements, so it's okay to deviate from it in this case?

@mischreiber Calling `super` doesn't cause the infinite loop. I intentionally avoid calling `super` because then it'd move focus up/down which we don't want for tab

Reply:
Even though FocusZone overrides native keyboarding behavior, I still think it has to respect the responder chain. In my view, FocusZone is effectively an NSView that does some extra handling in `keyDown` on arrow keys/tab to ensure focus goes where we like. If we determine that FocusZone doesn't have an action to do, it should call `super` so that the next responder handles the command. 

I think I don't understand why FocusZone needs to handle a keyboard event by explicitly doing nothing. I'll follow up with you offline.

Reply:
> ^ that's my question. What's the actual infinite loop that is happening? Is `nextValidKeyView == self` ?

Yes, `nextValidKeyView == self` is causing the infinite loop